src/test/java/com/jfrog/ide/idea/ui/BaseTreeTest.java

```diff
@@ -0,0 +1,98 @@
+package com.jfrog.ide.idea.ui;
+
+import org.jfrog.build.extractor.scan.DependenciesTree;
+import org.jfrog.build.extractor.scan.GeneralInfo;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * @author yahavi
+ */
+public class BaseTreeTest {
+
+    private BaseTree baseTree;
+    private DependenciesTree dependenciesTreeA;
+    private DependenciesTree dependenciesTreeB;
+    private DependenciesTree dependenciesTreeC;
+
+    @BeforeMethod
+    public void setUp() {
+        baseTree = new BaseTreeImpl();
+        dependenciesTreeA = createNode("a");
+        dependenciesTreeB = createNode("b");
+        dependenciesTreeC = createNode("c");
+    }
+
+    @Test
+    void oneProjectTest() {
+        // Append single project and check results
+        baseTree.appendProject(dependenciesTreeA);
+        Assert.assertEquals(baseTree.getModel().getRoot(), dependenciesTreeA);
+
+        // Reset tree
+        baseTree.reset();
+        Assert.assertNull(baseTree.getModel());
+
+        // Append single project again and check results
+        baseTree.appendProject(dependenciesTreeA);
+        Assert.assertEquals(baseTree.getModel().getRoot(), dependenciesTreeA);
+    }
+
+    @Test
+    void twoProjectsTest() {
```

Comment:
I suggest adding a description comment for each of the **Assert**s in this method. 
This should make it easier to understand.
Same suggestion for the *threeProjectsTest* method.